src/locutus_util/ontology_api_etl.py

```diff
@@ -92,25 +94,29 @@ def add_manual_ontologies():
 def add_monarch_ontologies():
     """Monarch API does not keep data on the ontologies themselves. Using a
     list of ontologies found in Monarch, backfill the information using the
-    data collected from the ols API.
+    data collected from the OLS API.
     
-    If some ontologies are not present in OLS add those with add_manual_ontologies.
+    If some ontologies are not present in OLS, add those with add_manual_ontologies.
     """
-    monarch_ontologies = ['CHEBI','ECTO','GO','HP','MAXO','MONDO','MP' \
-        'NBO','PATO','RO','SNOMED','UBERON']
+    monarch_ontologies = ['CHEBI', 'ECTO', 'GO', 'HP', 'MAXO', 'MONDO', 'MP', 
+                          'NBO', 'PATO', 'RO', 'SNOMED', 'UBERON']
+
+    # Create new entries based on existing OLS data
+    monarch_ols = []
+    for ontology in extracted_data:
+        if ontology['curie'] in monarch_ontologies:
+            new_entry = {
+                'api_url': MONARCH_API_BASE_URL,
+                'api_id': "monarch",
+                'api_name': "Monarch API",
+                'ontology_title': ontology['ontology_title'],
+                'ontology_code': ontology['ontology_code'],
+                'curie': ontology['curie'],
+                'system': ontology['system'],
+                'version': ontology['version'],
+            }
+            monarch_ols.append(new_entry)
     
-    # Filter out the entries whose 'curie' field is in monarch_ontologies
-    monarch_ols = [
-        entry for entry in extracted_data if entry['curie'] in monarch_ontologies
-    ]
-
-    # Update the required fields for the filtered data
-    for entry in monarch_ols:
-        entry['api_url'] = MONARCH_API_BASE_URL
-        entry['api_id'] = "monarch"
-        entry['api_name'] = "Monarch API"
-        entry['version'] = ""
-
     return monarch_ols
```

Comment:
I hope you don't mind a tag along. This logic was creating the issue we were seeing with the data so I went ahead and fixed it.

Reply:
looks good